atoma-proxy/src/sui/mod.rs

```diff
@@ -0,0 +1,232 @@
+use std::path::Path;
+
+use anyhow::Result;
+use atoma_sui::AtomaSuiConfig;
+use blake2::{
+    digest::generic_array::{typenum::U32, GenericArray},
+    Blake2b, Digest,
+};
+use serde_json::Value;
+use sui_keys::keystore::{AccountKeystore, Keystore};
+use sui_sdk::{
+    json::SuiJsonValue,
+    rpc_types::Page,
+    types::{
+        base_types::{ObjectID, SuiAddress},
+        crypto::EncodeDecodeBase64,
+        SUI_RANDOMNESS_STATE_OBJECT_ID,
+    },
+    wallet_context::WalletContext,
+    SuiClient,
+};
+use tracing::{info, instrument};
+
+const GAS_BUDGET: u64 = 5_000_000; // 0.005 SUI
+
+pub struct Sui {
+    wallet_ctx: WalletContext,
+    toma_wallet_id: Option<ObjectID>,
+    atoma_package_id: ObjectID,
```

Comment:
A potentially better strategy is to have a network `mainnet`, `devnet`, `testnet` and retrieve the right object ids depending on those. But this is good for now.